ci-operator/step-registry/idp/openldap/idp-openldap-commands.sh

```diff
@@ -0,0 +1,247 @@
+    temp_ldif=$(mktemp)
+    curl -o "$temp_ldif" https://gist.githubusercontent.com/dpuniaredhat/e8f98742abf6f44c9fda08b9b0a9ab45/raw/def42aa507c35c3212a69da64361ba65d522316b/gistfile1.txt --insecure
+
+    echo "Generating random passwords and updating temporary init.ldif file"
+    for i in $(seq 1 50); do
+        username="Person${i}"
+        password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+        users+="${username}:${password},"
+        sed -i "/dn: cn=${username},ou=people,ou=rfc2307,dc=example,dc=com/,/userPassword: random/s/userPassword: random/userPassword: ${password}/" "$temp_ldif"
+    done
+}
+
+#ldap root password var
+LDAP_SECRET=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+
+function configure_openldap() {
+    echo "Configuring OpenLDAP in OpenShift"
+    oc create ns ldap
+
+    echo "Creating a secret from the updated temporary init.ldif file"
+    oc create secret generic ldap-init-ldif --from-file=init.ldif="$temp_ldif" -n ldap
+    oc adm policy add-scc-to-user anyuid -z default -n ldap --context=admin
+
+    #Set ldap root password Secret
+    oc create secret generic ldap-admin-password --from-literal=OPENLDAP_ROOT_PASSWORD="${LDAP_SECRET}" -n ldap
+
+    # Define the LDAP deployment
+    cat <<EOF | oc apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ldap
+  namespace: ldap
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ldap
+  template:
+    metadata:
+      labels:
+        app: ldap
+    spec:
+      containers:
+      - name: ldap
+        image: quay.io/openshifttest/ldap:multiarch
+        env:
+        - name: OPENLDAP_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: ldap-admin-password
+              key: OPENLDAP_ROOT_PASSWORD
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - |
+                sleep 120
+                # Then run ldapadd command
+                ldapadd -x -H 'ldap://127.0.0.1:389' -D 'cn=Manager,dc=example,dc=com' -w \$OPENLDAP_ROOT_PASSWORD -f /tmp/init.ldif
+        volumeMounts:
+        - name: init-ldif-volume
+          mountPath: /tmp
+      volumes:
+      - name: init-ldif-volume
+        secret:
+          secretName: ldap-init-ldif
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ldap
+  namespace: ldap
+spec:
+  selector:
+    app: ldap
+  ports:
+  - protocol: TCP
+    port: 389
+    targetPort: 389
+EOF
+
+    # Below query to check record are updated on now in pod
+    # ldapsearch -x -H 'ldap://127.0.0.1:389' -D 'cn=Manager,dc=example,dc=com' -w $OPENLDAP_ROOT_PASSWORD -b 'dc=example,dc=com' '(objectclass=*)'
+
+    # Wait for LDAP pod to be running
+    echo "Checking for LDAP pod to be running"
+    count=0
+    max_attempts=10
+    while :; do
+        if [[ $(oc get pods -n ldap -o jsonpath='{.items[0].status.phase}') == 'Running' ]]; then
```

Comment:
To be robust, 'Running' may be dummy for `0/1` status. It is better to grep "ldap *1/1 *Running"?
```
NAME             READY   STATUS    RESTARTS   AGE   
xxxxxxxxxxxxxx   0/1     Running   0          1m
```